### ISSUE TITLE
Microarray builds failing with various warnings

### ISSUE THREAD

**Issue by malachig** [genome/gms#123]
Since the merge we seem to have a problem introduced here:

We are getting a lot of perl warnings, genome errors, and ultimately a crash:

```
ERROR: Failed to open bases file "/opt/gms/GMS1/fs/ams1102/info/model_data/2869585698/build106942997/bases/GL000001.1.bases".
Argument "GL000001.1" isn't numeric in numeric ne (!=) at /opt/gms/4DSHC42/sw/genome/lib/perl/Genome/Model/GenotypeMicroarray/GenotypeFile/ReadTsvAndAnnotate.pm line 148, <GEN10> line 127690172.

...
Command module died or returned undef.
Can't use string (".") as an ARRAY ref while "strict refs" in use at /opt/gms/4DSHC42/sw/genome/lib/perl/Genome/File/Vcf/Entry.pm line 779.
Genome::File::Vcf::Entry::to_string('Genome::File::Vcf::Entry=HASH(0x246e9fc8)') called at /opt/gms/4DSHC42/sw/genome/lib/perl/Genome/File/Vcf/Writer.pm line 35
```

Will talk to @tmooney about this one...

**Comment by malachig** [comment]
At first I thought this was related to issue #75.

But after talking to @tmooney it now seems clear that this is not specifically related to the above. Opening a separate issue.

I also opened an RT issue.  Here:
https://rt.gsc.wustl.edu/Ticket/Display.html?id=98226

The same issue seems to be occurring withing TGI.  Will need to resolve in master and merge the fix into gms-pub...

**Comment by malachig** [comment]
Microarray builds now succeeded.  However, many warnings are still thrown.  Furthermore a build that used to take ~30 minutes now takes ~24 hours...

**Comment by ebelter** [comment]
I will look into the warnings. The builds will get a little faster with the
next snapshot. They are using the dbsnp vcf to get more information. That
is taking a while. I am talking to Travis to try and further improve the
time. Part of the problem is they make a bunch of files that are just a
permutation of the genotype file.  Now the are using a very ineffiecnt Perl
vcf  reader.

-Eddie

**Comment by malachig** [comment]
I believe I was running my tests with the latest code committed to  master
as of Friday, so I'm guessing we are already using what will go out in the
next snapshot.

Is it possible that we also need an updated version of joinx?  It looks
like in the sGMS we have up to joinx1.7 but there is a joinx1.8 within
TGI...

**Comment by ebelter** [comment]
I believe that to be true. The new version has a bug fix. Forgot why we
have not upgraded. Allison knows more.

**Comment by apregier** [comment]
There is a remaining bug in joinx intersect vcf 1.8, so we haven't yet upped the "default" joinx version in the code.  This bug is being tracked in our internal RT (97906).

**Comment by obigriffith** [comment]
Has there been any progress on improving the run time of microarray 
builds?  Runs on sGMS this weekend still took about 22hours.  There are 
also tons of errors that look like:
Use of uninitialized value in numeric eq (==) at 
/opt/gms/D4F2D78/sw/genome/lib/perl/Genome/Model/GenotypeMicroarray/GenotypeFile/ReadUnannotatedCsv.pm 
line 230, <GEN10> line 52054864.

Perhaps there have been some recent changes to master that address this?

On 3/18/14 9:34 AM, apregier wrote:

> There is a remaining bug in joinx intersect vcf 1.8, so we haven't yet 
> upped the "default" joinx version in the code. This bug is being 
> tracked in our internal RT (97906).
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/genome/gms/issues/123#issuecomment-37939293.

## 

Obi L. Griffith, PhD
Research Assistant Professor
The Genome Institute, Washington University at St. Louis
Department of Medicine, Division of Oncology
Campus Box 8501
4444 Forest Park Ave
St. Louis, MO, 63108
Phone: 314-286-1274
www.obigriffith.org

---

This email message is a private communication. The information transmitted, including attachments, is intended only for the person or entity to which it is addressed and may contain confidential, privileged, and/or proprietary material. Any review, duplication, retransmission, distribution, or other use of, or taking of any action in reliance upon, this information by persons or entities other than the intended recipient is unauthorized by the sender and is prohibited. If you have received this message in error, please contact the sender immediately by return email and delete the original message from all computer systems. Thank you.

**Comment by ebelter** [comment]
We made many commits toward the end of the week. We have seen run times
under 20 minutes. When was the last time this was pulled from master?

**Comment by apregier** [comment]
We got a big speedup (as in faster than the previous runtime) with commit 640442370edf37569d on master.  It think I pushed late Thursday afternoon.  Do you have that merged in yet?
Eddie has pushed quite a few other changes as well.

**Comment by obigriffith** [comment]
Great. I'm guessing we do not have that merged in yet.

Avi - can you try merging in the latest microarray code and see if we 
get back to ~20-30min runtimes?

Obi

**Comment by gatoravi** [comment]
Are these all changes to lib/perl/Genome/File/Vcf/\* ? Are there any changes outside this folder too ?

**Comment by ebelter** [comment]
Allison's change was. I updated things in genotype microarray model dir.

**Comment by malachig** [comment]
I did a complete merge 9 days ago (March 16) here:
https://github.com/genome/gms-core/commit/40cf7ddeb36eb1a8f119524692a11792086d281d

That was the last one.  The last few complete merges went pretty smoothly.  Maybe we should try that instead of a selective merge?  Depends on how obvious the selective merge is and how many other things have changed that might introduce new problems...

**Comment by gatoravi** [comment]
yeah, might be worth trying a merge if there’s a stable snapshot with these changes. I’m checking that now.

**Comment by gatoravi** [comment]
there hasn't been a stable snapshot with @apregier 's commits in them. We'll have to wait for the next snapshot before merging in to avoid unstable builds.

**Comment by apregier** [comment]
This snapshot went out yesterday - you probably want to use the hotfix one (genome-3447-fix1)

**Comment by gatoravi** [comment]
yep, we just merged it in this morning ! Is a hotfix a git-tag ?

**Comment by apregier** [comment]
Sorry, I was wrong - hotfixes are not actually implemented as git repos, so there won't be a hotfix tag.  I guess you may want to do another merge when the next snapshot comes out so that you get the fix for the UsageLog warnings.

**Comment by malachig** [comment]
Many thanks to everyone who helped out on this especially @ebelter and @apregier 

We went ahead and merged gms-core master branch into the gms-pub branch here:
https://github.com/genome/gms-core/commit/61466329ada3b8f5ded848ec9d9cee53a2ecc8b2

Instead of waiting for the next snapshot to go out we determined from Jenkins tests that everything looked stable and merged at that time.

This has resolved the long running microarray build issue as well as the excessive warnings issue.  The workflow logs and work stage descriptions are also much nicer now.  

Closing this issue now.
